micromanager_gui/_gui_objects/_xyz_stages.py

```diff
@@ -1,69 +1,41 @@
-from pathlib import Path
+from qtpy.QtWidgets import QGroupBox, QHBoxLayout, QWidget
 
-from qtpy import QtWidgets as QtW
-from qtpy import uic
-from qtpy.QtCore import QSize
-from qtpy.QtGui import QIcon
+from .. import _core
+from .._core_widgets._stage_widget import StageWidget
 
-ICONS = Path(__file__).parent.parent / "icons"
-
-
-class MMStagesWidget(QtW.QWidget):
-
-    MM_XYZ_STAGE = str(Path(__file__).parent / "mm_xyz_stage.ui")
-
-    # The MM_XYZ_STAGE above contains these objects:
-
-    XY_groupBox: QtW.QGroupBox
-    xy_device_comboBox: QtW.QComboBox
-    xy_step_size_SpinBox: QtW.QSpinBox
-    y_up_Button: QtW.QPushButton
-    y_down_Button: QtW.QPushButton
-    left_Button: QtW.QPushButton
-    right_Button: QtW.QPushButton
-
-    Z_groupBox: QtW.QGroupBox
-    z_step_size_doubleSpinBox: QtW.QDoubleSpinBox
-    focus_device_comboBox: QtW.QComboBox
-    up_Button: QtW.QPushButton
-    down_Button: QtW.QPushButton
-
-    offset_Z_groupBox: QtW.QGroupBox
-    offset_device_comboBox: QtW.QComboBox
-    offset_z_step_size_doubleSpinBox: QtW.QDoubleSpinBox
-    offset_up_Button: QtW.QPushButton
-    offset_down_Button: QtW.QPushButton
-
-    x_lineEdit: QtW.QLineEdit
-    y_lineEdit: QtW.QLineEdit
-    z_lineEdit: QtW.QLineEdit
-
-    snap_on_click_checkBox: QtW.QCheckBox
 
+class MMStagesWidget(QWidget):
     def __init__(self, parent=None):
         super().__init__(parent)
-        uic.loadUi(self.MM_XYZ_STAGE, self)
-
-        # button icons
-        for attr, icon in [
-            ("left_Button", "left_arrow_1_green.svg"),
-            ("right_Button", "right_arrow_1_green.svg"),
-            ("y_up_Button", "up_arrow_1_green.svg"),
-            ("y_down_Button", "down_arrow_1_green.svg"),
-            ("up_Button", "up_arrow_1_green.svg"),
-            ("down_Button", "down_arrow_1_green.svg"),
-            ("offset_up_Button", "up_arrow_1_green.svg"),
-            ("offset_down_Button", "down_arrow_1_green.svg"),
-        ]:
-            btn = getattr(self, attr)
-            btn.setIcon(QIcon(str(ICONS / icon)))
-            btn.setIconSize(QSize(30, 30))
-
-
-if __name__ == "__main__":
-    import sys
-
-    app = QtW.QApplication(sys.argv)
-    win = MMStagesWidget()
-    win.show()
-    sys.exit(app.exec_())
+        self.setLayout(QHBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().setSpacing(6)
+        self._mmc = _core.get_core_singleton()
+        self._on_cfg_loaded()
+        self._mmc.events.systemConfigurationLoaded.connect(self._on_cfg_loaded)
+
+    def _on_cfg_loaded(self):
+        self._clear()
+        if dev := self._mmc.getXYStageDevice():
```

Comment:
I think we should provide an object for generic `StageWidgets` and then a subclass that follows the default stage widget. Like what I did in #150 (in spirit if not in implementation)

Reply:
that's primarily what this PR is doing.  See [`micromanager_gui/_core_widgets/_stage_widget.py`](https://github.com/tlambert03/napari-micromanager/pull/142/files#diff-2c246c921e3e8559c18cb076e6e01fbf676d34a9751d7a68dc9322187e790c7d)

this file here is just for the few things that are special about napari micromanager (like the QCollapsible).  hence why it's in gui_objects :)

Reply:
and this one arranges the generic stage widget into a multi widget.  That could also happen in core, but the primary widget here is `class StageWidget(QWidget)` in that other file